core/vm/vmcontext/context.go

```diff
@@ -4,16 +4,37 @@ import (
 	"math/big"
 
 	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/consensus"
 	"github.com/celo-org/celo-blockchain/consensus/istanbul"
 	"github.com/celo-org/celo-blockchain/contracts"
 	"github.com/celo-org/celo-blockchain/contracts/reserve"
 	"github.com/celo-org/celo-blockchain/core/types"
 	"github.com/celo-org/celo-blockchain/core/vm"
 	"github.com/celo-org/celo-blockchain/log"
+	"github.com/celo-org/celo-blockchain/params"
 )
 
+// chainContext defines methods required to build a context
+// a copy of this exist on core.ChainContext (needed to break dependency)
+type chainContext interface {
+	// Engine retrieves the blockchain's consensus engine.
+	Engine() consensus.Engine
+
+	// GetHeader returns the hash corresponding to the given hash and number.
+	GetHeader(common.Hash, uint64) *types.Header
+
+	// GetHeaderByNumber returns the hash corresponding number.
+	// FIXME: Use of this function, as implemented, in the EVM context produces undefined behavior
+	// in the pressence of forks. A new method needs to be created to retrieve a header by number
```

Comment:
Same as in `core`: either fix the typo in `pressence` or remove the FIXME since it's about forks